Add `onlyAnnotatedFor` flag (#1290)
eisop/checker-framework: docs/manual/annotating-libraries.tex

conservative defaults
(see Section~\ref{defaults-classfile}) for any type use with no explicit
user-written annotation, \emph{and} the checker issues no warnings.

> \code{-AonlyAnnotatedFor} is an option that only suppresses warnings outside of the \<@AnnotatedFor> annotation,

Reviewer: ```suggestion
The command-line argument \code{-AonlyAnnotatedFor} can be used to  suppress warnings and errors outside of the scope of an \<@AnnotatedFor> annotation,
```

Reviewer: Tested latest commit against [Spring-Framework](https://github.com/spring-projects/spring-framework) with expected results.

> The command-line argument \code{-AonlyAnnotatedFor} can be used to  suppress warnings and errors outside of the scope of an \<@AnnotatedFor> annotation,

Reviewer 2: ```suggestion
The command-line argument \code{-AonlyAnnotatedFor} can be used to  suppress errors and warnings outside of the scope of an \<@AnnotatedFor> annotation,
```

> but does not change the default qualifier for source code (See Section~\ref{aonlyAnnotatedFor}).

Reviewer 2: ```suggestion
but does not change the default qualifier for source code (See Section~\ref{AonlyAnnotatedFor}).
```

> but does not change the default qualifier for source code (See Section~\ref{aonlyannotatedfor}).

Reviewer 2: ```suggestion
but does not change the default qualifiers for source code (See Section~\ref{aonlyannotatedfor}).
```
